src/supermarket/spec/controllers/api/v1/cookbook_uploads_controller_spec.rb

```diff
@@ -244,6 +244,21 @@
       end
     end
 
+    context 'when there is only one version of the cookbook remaining' do
+      before { auto_authorize!(Cookbook, 'destroy') }
+
+      it 'returns a 409 with informative error messages' do
+        versions = CookbookVersion.all.pluck(:version)
+        versions.each do |version|
+          delete :destroy_version, cookbook: cookbook, version: version, format: :json
+        end
+
+        expect(response.status.to_i).to eql(409)
+        expect(JSON.parse(response.body)["error_code"]).to eq('CONFLICT')
+        expect(JSON.parse(response.body)["error_messages"][0]).to eq('You may not DELETE the only version of a cookbook. A cookbook must have at least one version.')
```

Comment:
I recommend we test for `error` presence in the response body because we _know_ that is the message being used by the knife CLI. 

And what are your thoughts on testing that it matches the translation for the message, rather than the string repeated here? I waffle between making the test declarative but brittle with the exact string or making the test less brittle but maybe more indirect by using `I18n.t('api.error_messages.only_cookbook_version')`.

Yey! Tests!